packages/docs/docs/docs/linking.md

```diff
@@ -1,127 +1,281 @@
+
+<!-- We can split the following into a separate tutorial if this one becomes too long -->
+
+## Writing the exchange contract
+
+Our exchange contract will need to store the token contract address and the exchange rate in its state. We will set these values during initialization, when we deploy our contract.
+
+In order to support contract upgrades, ZeppelinOS [does not allow the usage of Solidity's `constructor`s](pattern#the-constructor-caveat). Instead, we need to use _initializers_. An initializer is just a regular Solidity function, with an additional check to ensure that it can be called only once. To make coding initializers easy, ZeppelinOS provides a base `Initializable` contract, that includes an `initializer` modifier that takes care of this. You will first need to install the package that provides that contract:
+
+```console
+npm install zos-lib@2.4.0
+```
+
+Now, let's write our exchange contract using an _initializer_ to set its initial state:
 
 ```solidity
 pragma solidity ^0.5.0;
 
-import "openzeppelin-eth/contracts/token/ERC721/ERC721Mintable.sol";
+// Import base Initializable contract
+import "zos-lib/contracts/Initializable.sol";
+
+// Import interface and library from OpenZeppelin contracts
+import "openzeppelin-eth/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-eth/contracts/math/SafeMath.sol";
 
-contract MyLinkedContract {
-  ERC721Mintable private _token;
+contract TokenExchange is Initializable {
+  using SafeMath for uint256;
 
-  function setToken(ERC721Mintable token) external {
-    require(address(token) != address(0));
-    _token = token;
+  // Contract state: exchange rate and token
+  uint256 public rate;
+  IERC20 public token;
+
+  // Initializer function (replaces constructor)
+  function initialize(uint256 _rate, IERC20 _token) public initializer {
+    rate = _rate;
+    token = _token;
   }
-}
+
+  // Send tokens back to the sender using predefined exchange rate
+  function() external payable {
+    uint256 tokens = msg.value.mul(rate);
+    token.transfer(msg.sender, tokens);
+  }
+} 
 ```
 
-One thing to notice is that instead of importing `openzeppelin-solidity` we are
-importing `openzeppelin-eth`. This is the name of the OpenZeppelin EVM package,
-the one we have to use if we want to reuse the package already deployed.
-For more information, see
-[this article which explains the difference between openzeppelin-solidity and openzeppelin-eth](https://blog.zeppelin.solutions/getting-started-with-openzeppelin-eth-a-new-stable-and-upgradeable-evm-package-576fb37297d0#125e).
+Note the usage of the `initializer` modifier in the `initialize` method. This guarantees that, after we have deployed our contract, no one can call into that function again to alter the token or the rate.
 
-Now, let's link our project to the openzeppelin-eth package by running:
+Let's now create and initialize our new `TokenExchange` contract:
 
 ```console
-npx zos link openzeppelin-eth
+$ zos create
```

Comment:
"npx zos create" or "zos create"

Reply:
Good catch! I missed to change the installation instructions in `linking`